### PR TITLE
Add PinUpdate, make DedupAndCalculatePinSize non-class function

### PR DESCRIPTION
**Pull request by bonedaddy**

### CODE REVIEW

File: rtfs.go [hunk 1]
```diff
@@ -102,6 +102,24 @@ func (im *IpfsManager) Pin(hash string) error {
...
 }
 
+// PinUpdate is used to update one pin to another, while making sure all objects
+// in the new pin are local, followed by removing the old pin.
+//
+// This is an optimized version of pinning the new content, and then removing the
+// old content.
+//
+// returns the new pin path
+func (im *IpfsManager) PinUpdate(fromPath, toPath string) (string, error) {
```
Comment (bobheadxi) [inline rtfs.go]: do you mean `PinMigrate` or `PinMove`?

Comment (bobheadxi) [reply rtfs.go]: I would also make the naming simpler 
```suggestion
func (im *IpfsManager) PinUpdate(from, to string) (string, error) {
```

Comment (bonedaddy) [reply rtfs.go]: No the IPFS command is `ipfs pin update` so I felt it best to stick with the naming

File: rtfs.go [hunk 2]
```diff
@@ -102,6 +102,24 @@ func (im *IpfsManager) Pin(hash string) error {
...
+	out, err := im.shell.PinUpdate(fromPath, toPath)
+	if err != nil {
+		return "", err
+	}
+	if len(out) == 0 {
+		return "", errors.New("failed to retrieve new pin paths")
+	}
+	return out["Pins"][1], nil
```
Comment (bobheadxi) [inline rtfs.go:117]: we should check for length here too, since this might panic

File: rtfs.go [hunk 3]
```diff
@@ -159,11 +177,13 @@ func (im *IpfsManager) SwarmConnect(ctx context.Context, addrs ...string) error
 	return im.shell.SwarmConnect(ctx, addrs...)
 }
 
+// non-class functions
+
 // DedupAndCalculatePinSize is used to remove duplicate refers to objects for a more accurate pin size cost
 // it returns the size of all refs, as well as all unique references
-func (im *IpfsManager) DedupAndCalculatePinSize(hash string) (int64, []string, error) {
+func DedupAndCalculatePinSize(hash string, im Manager) (int64, []string, error) {
```
Comment (bobheadxi) [inline rtfs.go]: maybe move this to another file?

**Review by bobheadxi — DISMISSED** [review]

File: rtfs.go (hunk 1 above)
Comment (bobheadxi) [reply rtfs.go]: ah I see

File: rtfs.go (hunk 3 above)
Comment (bonedaddy) [reply rtfs.go]: Good point

File: rtfs.go (hunk 2 above)
Comment (bonedaddy) [reply rtfs.go:117]: Good catch

**Comment by bonedaddy** [comment]
Doing this in another branch without gx

**Pull request closed without merging**